### ISSUE TITLE
required condition is false: IsFormatSampleRateAndChannelCountValid(format)

### ISSUE THREAD

**Issue by Muthu011** [react-native-voice/voice#388]
{error: {message: "required condition is false: IsFormatSampleRateAndChannelCountValid(format)", code: "start_recording"}}
overrideMethod — index.bundle:39716 

Hi , I am getting this error in ios real device. When ever i try to click mic button i am triggering speech start handler and suddenly error is captured in speech error handler. The above one is the error message. Someone please help to fix the problem. It happens only in the ios but in android it is working fine.

**Comment by Muthu011** [comment]
this error comes only when the tts is also installed in the project. still haven't figured it out someone find fix or known the fix please do post thanks in advance.

**Comment by NitinSidhuOfficial** [comment]
any update on this issue?

**Comment by RakshithCodewave** [comment]
I'm facing the same issue.

**Comment by arifrohmanhakim** [comment]
any update on this issue?

**Comment by Ahmed-Amin-Qoad** [comment]
I'm facing the same issue.

**Comment by chace20** [comment]
same issue still not resolved

**Comment by craig1123** [comment]
I received this error in the iPhone simulator from my mac mini. Turns out, I just needed to connect a mic to my computer and the error went away.

**Comment by bibhuti9** [comment]
I'm facing the same issue, please help.

**Comment by Th1nhNg0** [comment]
Same issue till 2025

**Comment by mkamals1989** [comment]
> I think there is a conflict with react-native-video

I faced the same issue. If you are using  react-native-video in combination with this lib and you are managing the audio by yourself then you have to explicitly set this prop in <Video /> component disableAudioSessionManagement={true}.

**Comment by YashYash** [comment]
Is there a solution to this problem? This is blocking a production release for us :(.

**Comment by d-baumann** [comment]
I had the same issue, downgrading react-native-video from 6.15.0 to 6.10.2 fixed it for me

**Comment by PriybhashSoni** [comment]
> > I think there is a conflict with react-native-video
> 
> I faced the same issue. If you are using react-native-video in combination with this lib and you are managing the audio by yourself then you have to explicitly set this prop disableAudioSessionManagement in Video Component.
> 
> `<Video component disableAudioSessionManagement={true} />`



> > I think there is a conflict with react-native-video
> 
> I faced the same issue. If you are using react-native-video in combination with this lib and you are managing the audio by yourself then you have to explicitly set this prop disableAudioSessionManagement in Video Component.
> 
> `<Video component disableAudioSessionManagement={true} />`

it is working thanks @mkamals1989

**Comment by dprajapati1179** [comment]
so far any updates @safaiyeh

**Comment by Sapnarawat1** [comment]
yes the issue is in new RN version audioSessionManager of react-native-video ,
update your updateAudioSessionConfiguration function in AudioSessionManager.swift file with this.

  func updateAudioSessionConfiguration() {
        let hasVideoViews = !videoViews.allObjects.isEmpty
        let isAnyPlayerPlaying = videoViews.allObjects.contains { view in
            return !view.isMuted() && view._player != nil && view._player?.rate != 0
        }
        // Only keep audio session active if there are video views AND (players playing OR remote control events active)
        if hasVideoViews && (isAnyPlayerPlaying || remoteControlEventsActive) {
            activateAudioSession()
            configureAudioSession()
        } else {
            // No video views or no players playing, deactivate and reset audio session
            deactivateAudioSession()
        }
    }

**Comment by dprajapati1179** [comment]
> yes the issue is in new RN version audioSessionManager of react-native-video , update your updateAudioSessionConfiguration function in AudioSessionManager.swift file with this.
> 
> func updateAudioSessionConfiguration() { let hasVideoViews = !videoViews.allObjects.isEmpty let isAnyPlayerPlaying = videoViews.allObjects.contains { view in return !view.isMuted() && view._player != nil && view._player?.rate != 0 } // Only keep audio session active if there are video views AND (players playing OR remote control events active) if hasVideoViews && (isAnyPlayerPlaying || remoteControlEventsActive) { activateAudioSession() configureAudioSession() } else { // No video views or no players playing, deactivate and reset audio session deactivateAudioSession() } }

I will try this just confirm that this is a patch to react-native-video , Right ?

**Comment by safaiyeh** [comment]
This project has been deprecated and archived. Please migrate to [`expo-speech-recognition`](https://github.com/jamsch/expo-speech-recognition). Thank you for your contributions! ❤️
